### PR TITLE
Boundary screen

### PR DESCRIPTION
**Pull request by abishekTa-egov**
Campaign Boundary screen for summary screen, ready to take data.

## Release Notes

- **New Features**
  - Introduced the `CampaignBoundary` component for enhanced campaign boundary management.
  - Added a new card for campaign boundary information in the Summary Screen.
  - Implemented a new `SubBoundaryView` component to display boundary details.
  - Added a new `selectedData` structure for geographical entities in the Sample page.

- **Bug Fixes**
  - Updated loading state handling in the `CampaignDetails` component for improved user experience.

- **Style**
  - Enhanced styles with new classes for better component presentation in the micro UI framework.
  - Updated CSS stylesheet version to ensure the latest styles are applied.

- **Chores**
  - Updated the CSS stylesheet version in the HTML document.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+        } else {
+            updatedBoundaryStatus[ind] = true;
+
+        }
+        // Set the updated array in the state
+        setBoundaryStatus(updatedBoundaryStatus);
+        // Set showAllItems to true to show all items
+    };
+
+    const selectedData = [
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js:29]: Remove this

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+        for (const ind in bHierarchy) {
+            if (parent_group[bHierarchy[ind]] && parent_group[bHierarchy[ind]].length > 2) {
+                boundaryStatus[ind] = true
+
+            } else {
+                boundaryStatus[ind] = false
+
+            }
+        }
```
[the remark this answers is not in the document]
Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js:412]: @abishekTa-egov Update here

**Review by nipunarora-eGov — CHANGES_REQUESTED** [review]
@abishekTa-egov check codeRabbit comments and rectify them

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js [hunk 3]
```diff
@@ -0,0 +1,520 @@
+import React, { useState, useEffect, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { useMyContext } from "../utils/context";
+import SubBoundaryView from "./subBoundaryView";
+import HeaderComp from "./HeaderComp";
+import { Card } from "@egovernments/digit-ui-components";
+import BoundaryKpi from "./BoundaryKpi";
+import { unstable_batchedUpdates } from "react-dom";
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js:8]: What is this import?

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/Sample.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        "name": "MICROPLAN_MO_05_05_GREENVILLE",
+        "type": "District",
+        "parent": "MICROPLAN_MO_05_SINOE",
+        "isRoot": false,
+        "includeAllChildren": false
+    },
+    {
+        "code": "MICROPLAN_MO_05_04_GBLONEE",
+        "name": "MICROPLAN_MO_05_04_GBLONEE",
+        "type": "District",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/Sample.js:188]: remove this file if not required

File: health/micro-ui/web/micro-ui-internals/packages/css/src/components/microplan.scss [hunk 5]
```diff
@@ -96,4 +96,17 @@
     background-color: #fbe9e6;
     border-color: #dc5a32;
   }
+
+  .subBoundarycomp-container{
+    display:"inline-flex";
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/css/src/components/microplan.scss:101]: this css is wrong, remove quotes

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js (hunk 1 above)
Comment (abishekTa-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js]: I don't understand what to remove

**Review by abishekTa-egov — COMMENTED** [review]
I have resolved all the comments

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+
+        }
+        // Set the updated array in the state
+        setBoundaryStatus(updatedBoundaryStatus);
+        // Set showAllItems to true to show all items
+    };
+
+    const selectedData1 = customProps?.sessionData?.BOUNDARY?.boundarySelection?.selectedData;
+
+    const selectedData=[
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/CampaignBoundary.js:31]: can you remove this hardcoded data

File: health/micro-ui/web/public/index.html [hunk 7]
```diff
@@ -12,7 +12,11 @@
   <link rel="stylesheet" href="https://unpkg.com/@egovernments/digit-ui-components-css@0.0.2-beta.34/dist/index.css" />
   <!-- added below css for hcm-workbench module inclusion-->
 
+<<<<<<< HEAD
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/public/index.html:15]: Resolve this change

**Review by nipunarora-eGov — APPROVED** [review]

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**